SupyML/plugin.py

```diff
@@ -136,22 +139,27 @@ def _processNode(self, node, variables, nested=True):
 
         if isinstance(node, minidom.Text):
             return node.data
+        if node.nodeName=='loop':
+                return self._processLoop(node, variables)
+        if node.nodeName=='var':
+                return self._processVar(node, variables)
+        if node.nodeName=='set':
+                return self._processSet(node, variables)
+        if node.nodeName=='if':
+                return self._processIf(node, variables)
+
         output = node.nodeName + ' '
         for childNode in node.childNodes:
             self._startNode()
             if not repr(node) == repr(childNode.parentNode):
                 continue
-            if childNode.nodeName == 'loop':
-                output += self._processLoop(childNode, variables)
-            elif childNode.nodeName == 'if':
-                output += self._processId(childNode, variables)
-            elif childNode.nodeName == 'var':
-                output += self._processVar(childNode, variables)
-            elif childNode.nodeName == 'set':
-                output += self._processSet(childNode, variables)
-            else:
-                output += self._processNode(childNode, variables) or ''
+            output += self._processNode(childNode, variables) or ''
+        log.info("SupyML: node statement: %s"%(str(output)))
         value = self._run(output, nested)
+        if type(value) is ircmsgs.IrcMsg :
+            log.warn("SupyML: resulting msg "+str(value)+" is an irc command. Sending to network.")
+            self._irc.sendMsg(value)
```

Comment:
use `self._irc.queueMsg` instead; `self._irc.sendMsg` should only be used for high-priority messages.

Reply:
I originally had queueMsg() here, but for some reason it didn't work. In some deeply nested cases the messages never got executed. (I wrote a script to voice all users in a channel with a foreach loop)  I might have to debug that a bit more.

Reply:
changed the way the Proxy is implemented, so non direct replies are sent through the regular nested proxy in supybot/callback.py